tools/ci_status_report.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+import os
+import enum
+import argparse
+import collections
+import datetime
+import dataclasses
+import itertools
+
+import requests
+import matplotlib.pyplot as plt
+
+
+PROW_JOBS_STATUSES = "https://prow.ci.openshift.org/prowjobs.js?omit=annotations,labels,decoration_config,pod_spec"
+SLACK_FILES_UPLOAD = "https://slack.com/api/files.upload"
+TRACKED_REPOSITORIES = list(itertools.chain(
+    itertools.product(
+        ("openshift",), (
+            "assisted-test-infra",
+            "assisted-service",
+            "assisted-image-service",
+            "assisted-installer",
+            "assisted-installer-agent",
+        )),
+    itertools.product(
+        ("openshift-assisted",), (
+            "assisted-installer-deployment",
+            "assisted-events-scrape",
+        )),
+))
+
+
+class State(enum.Enum):
+    TRIGGERED = "triggered"  # job created but not yet scheduled
+    PENDING = "pending"  # job is currently running
+    SUCCESS = "success"  # job completed without any error
+    FAILURE = "failure"  # job completed with errors
+    ABORTED = "aborted"  # prow killed the job early (new commit pushed, perhaps)
+    ERROR = "error"  # job could not schedule (bad config, perhaps)
+
+
+class JobType(enum.Enum):
+    PRESUBMIT = "presubmit"
+    POSTSUBMIT = "postsubmit"
+    PERIODIC = "periodic"
+    BATCH = "batch"
+
+
+STATE_TO_COLOR = {
+    State.PENDING: "#FFD166",
+    State.ABORTED: "#118AB2",
+    State.FAILURE: "#EF476F",
+    State.ERROR: "pink",
+    State.SUCCESS: "#06D6A0",
+}
+
+
+@dataclasses.dataclass
+class Job:
+    type: JobType
+    name: str
+    organization: str
+    repository: str
+    base_ref: str
+    state: State
+    url: str
+    start_time: datetime.datetime
+    completion_time: datetime.datetime
+
+
+def filter_jobs(response):
+    for document in response.json()["items"]:
+        all_refs = document["spec"].get("extra_refs", [])
+
+        refs = document["spec"].get("refs")
+        if refs is not None:
+            all_refs.append(refs)
+
+        for ref in all_refs:
+            organization = ref["org"]
+            repository = ref["repo"]
+            base_ref = ref["base_ref"]
+            if (organization, repository) in TRACKED_REPOSITORIES:
+                break
+
+        else:
+            continue
+
+        job = Job(
+            type=JobType(document["spec"]["type"]),
+            name=document["spec"]["job"],
+            organization=organization,
+            repository=repository,
+            base_ref=base_ref,
+            state=State(document["status"]["state"]),
+            url=document["status"]["url"],
+            start_time=document["status"]["startTime"],
+            completion_time=document["status"].get("completionTime"),
+        )
+
+        if (job.organization, job.repository) in TRACKED_REPOSITORIES:
```

Comment:
Isn't this condition redundant?

Reply:
no, not at all
for some reason, I couldn't query this API to get just jobs for specific repos/organizations. So it basically brings everything and we filter our stuff from it

Reply:
I should have emphasized - you already check this condition in 

```python
        for ref in all_refs:
            organization = ref["org"]
            repository = ref["repo"]
            base_ref = ref["base_ref"]
            if (organization, repository) in TRACKED_REPOSITORIES:
                break

        else:
            continue
```

Reply:
oh yes, you're right
I'll remove it